tensorflow/core/platform/cloud/retrying_file_system.cc

```diff
@@ -37,55 +42,81 @@ bool IsRetriable(Status status) {
   }
 }
 
-Status CallWithRetries(const std::function<Status()>& f) {
-  int attempts = 0;
+void WaitBeforeRetry(const int delay_seconds) {
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<> dist(0, 999);
+
+  const int64 random_ms = dist(gen);
+  const int64 delay_ms = std::min(delay_seconds * 1000 + random_ms,
+                                  kMaximumBackoffMilliseconds);
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(delay_ms));
```

Comment:
Please use Env::SleepForMicroseconds instead:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/env.h#L220

There are internal reasons why we need to use a non-standard sleep call.